src/mbpfan.c

```diff
@@ -573,12 +584,31 @@ void mbpfan()
 {
     int old_temp, new_temp, fan_speed, steps;
     int temp_change;
+    int err;
     
     sensors = retrieve_sensors();
     fans = retrieve_fans();
 
     retrieve_settings(NULL, fans);
     
+    if (intel_pstate_control) {
+        if (!intel_pstate_is_available()) {
+            mbp_log(LOG_ERR, "Intel pstate control is requested but not available");
+            exit(EXIT_FAILURE);
+        } else {
+            intel_pstate = (t_intel_pstate*)malloc(sizeof(t_intel_pstate));
+            if (!intel_pstate) {
```

Comment:
We don't check `malloc` failure in the rest of the code so let's not bother here.

Reply:
Will do.